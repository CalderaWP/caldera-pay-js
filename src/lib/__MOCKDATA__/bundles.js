export default [{"id":48255,"date":"2017-03-01T10:18:24","date_gmt":"2017-03-01T15:18:24","guid":{"rendered":"https://test-calderaformscom.pantheonsite.io/?post_type=download&#038;p=48255"},"modified":"2018-08-03T11:09:59","modified_gmt":"2018-08-03T15:09:59","slug":"caldera-forms-enterprise-bundle","status":"publish","type":"download","link":"https://calderaformscom.lndo.site/downloads/caldera-forms-enterprise-bundle/","title":{"rendered":"Caldera Forms Pro Agency"},"content":{"rendered":"","protected":false},"excerpt":{"rendered":"","protected":false},"author":1,"featured_media":0,"template":"","caldera_admin_help":[],"caldera_doc_overview":[],"calderaPay":{"prices":{"ID":48255,"price":"549.89","free":false,"options":[],"includes":[],"isBundle":false,"addToCart":"https://calderaformscom.lndo.site/checkout?edd_action=add_to_cart&download_id=48255"},"featuredImage":{"ID":0,"html":"<img width=\"48\" height=\"64\" src=\"https://calderaformscom.lndo.site/wp-includes/images/media/default.png\" class=\"attachment-large size-large\" alt=\"\" />","url":false,"width":"1200","height":"437","srcset":false,"sizes":false,"alt":false},"bundle":{"isBundle":true,"includes":["6108","21068","6077","1923","1223","37221","43154","552","1934","1218","1950","607","3221","5070","21066","6151","8241","333","3688","2465","20522","1940","1918","565","2516","55330","56332","2771","24292","64101_6","561"]}},"cfdc_image":{"ID":0,"html":"<img width=\"48\" height=\"64\" src=\"https://calderaformscom.lndo.site/wp-includes/images/media/default.png\" class=\"attachment-large size-large\" alt=\"\" />","url":"https://calderaforms.com/wp-content/uploads/edd/2016/11/mark-viewed.png","width":"1200","height":"437","srcset":false,"sizes":false,"alt":false},"cfdc_prices":{"ID":48255,"price":"549.89","free":false,"options":[]},"_links":{"self":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download/48255"}],"collection":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download"}],"about":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/types/download"}],"author":[{"embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/users/1"}],"version-history":[{"count":4,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download/48255/revisions"}],"predecessor-version":[{"id":109537,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download/48255/revisions/109537"}],"wp:attachment":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/media?parent=48255"}],"wp:term":[{"taxonomy":"caldera_admin_help","embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/caldera_admin_help?post=48255"},{"taxonomy":"caldera_doc_overview","embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/caldera_doc_overview?post=48255"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}},{"id":20520,"date":"2016-06-07T21:52:22","date_gmt":"2016-06-08T01:52:22","guid":{"rendered":"https://test-calderaformscom.pantheonsite.io/?post_type=download&#038;p=20520"},"modified":"2018-04-30T10:57:10","modified_gmt":"2018-04-30T14:57:10","slug":"caldera-forms-developer-bundle","status":"publish","type":"download","link":"https://calderaformscom.lndo.site/downloads/caldera-forms-developer-bundle/","title":{"rendered":"Caldera Forms Pro Individual"},"content":{"rendered":"","protected":false},"excerpt":{"rendered":"<p>Get started with Caldera Forms + PayPal, partial form submissions, and user registration. </p>\n","protected":false},"author":1,"featured_media":16735,"template":"","caldera_admin_help":[],"caldera_doc_overview":[],"calderaPay":{"prices":{"ID":20520,"price":"164.89","free":false,"options":[],"includes":[],"isBundle":false,"addToCart":"https://calderaformscom.lndo.site/checkout?edd_action=add_to_cart&download_id=20520"},"featuredImage":{"ID":16735,"html":"<img width=\"660\" height=\"214\" src=\"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg\" class=\"attachment-large size-large\" alt=\"Caldera Forms Banner\" srcset=\"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg 772w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-300x97.jpg 300w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-768x249.jpg 768w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-632x205.jpg 632w\" sizes=\"(max-width: 660px) 100vw, 660px\" />","url":"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg","width":"1200","height":"437","srcset":"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-300x97.jpg 300w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-768x249.jpg 768w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-632x205.jpg 632w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg 772w","sizes":"(max-width: 150px) 100vw, 150px","alt":"Caldera Forms Banner"},"bundle":{"isBundle":true,"includes":["1918","1923","1940","3221","4960","62145","561","607","5070","64101_4"]}},"cfdc_image":{"ID":16735,"html":"<img width=\"660\" height=\"214\" src=\"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg\" class=\"attachment-large size-large\" alt=\"Caldera Forms Banner\" srcset=\"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg 772w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-300x97.jpg 300w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-768x249.jpg 768w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-632x205.jpg 632w\" sizes=\"(max-width: 660px) 100vw, 660px\" />","url":"https://calderaforms.com/wp-content/uploads/edd/2016/11/mark-viewed.png","width":"1200","height":"437","srcset":"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-300x97.jpg 300w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-768x249.jpg 768w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-632x205.jpg 632w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg 772w","sizes":"(max-width: 150px) 100vw, 150px","alt":"Caldera Forms Banner"},"cfdc_prices":{"ID":20520,"price":"164.89","free":false,"options":[]},"_links":{"self":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download/20520"}],"collection":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download"}],"about":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/types/download"}],"author":[{"embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/users/1"}],"version-history":[{"count":5,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download/20520/revisions"}],"predecessor-version":[{"id":110798,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download/20520/revisions/110798"}],"wp:featuredmedia":[{"embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/media/16735"}],"wp:attachment":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/media?parent=20520"}],"wp:term":[{"taxonomy":"caldera_admin_help","embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/caldera_admin_help?post=20520"},{"taxonomy":"caldera_doc_overview","embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/caldera_doc_overview?post=20520"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}},{"id":20518,"date":"2016-05-29T19:03:40","date_gmt":"2016-05-29T23:03:40","guid":{"rendered":"https://test-calderaformscom.pantheonsite.io/?post_type=download&#038;p=20518"},"modified":"2018-04-19T02:07:53","modified_gmt":"2018-04-19T06:07:53","slug":"caldera-forms-agency-bundle","status":"publish","type":"download","link":"https://calderaformscom.lndo.site/downloads/caldera-forms-agency-bundle/","title":{"rendered":"Caldera Forms Pro Advanced"},"content":{"rendered":"","protected":false},"excerpt":{"rendered":"<p>Every Caldera Forms add-on, 20 sites!</p>\n","protected":false},"author":1,"featured_media":16735,"template":"","caldera_admin_help":[],"caldera_doc_overview":[],"calderaPay":{"prices":{"ID":20518,"price":"274.89","free":false,"options":[],"includes":[],"isBundle":false,"addToCart":"https://calderaformscom.lndo.site/checkout?edd_action=add_to_cart&download_id=20518"},"featuredImage":{"ID":16735,"html":"<img width=\"660\" height=\"214\" src=\"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg\" class=\"attachment-large size-large\" alt=\"Caldera Forms Banner\" srcset=\"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg 772w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-300x97.jpg 300w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-768x249.jpg 768w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-632x205.jpg 632w\" sizes=\"(max-width: 660px) 100vw, 660px\" />","url":"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg","width":"1200","height":"437","srcset":"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-300x97.jpg 300w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-768x249.jpg 768w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-632x205.jpg 632w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg 772w","sizes":"(max-width: 150px) 100vw, 150px","alt":"Caldera Forms Banner"},"bundle":{"isBundle":true,"includes":["1918","1923","1223","552","561","565","607","3221","5070","2771","62145","1940","4960","43154","64101_5"]}},"cfdc_image":{"ID":16735,"html":"<img width=\"660\" height=\"214\" src=\"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg\" class=\"attachment-large size-large\" alt=\"Caldera Forms Banner\" srcset=\"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg 772w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-300x97.jpg 300w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-768x249.jpg 768w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-632x205.jpg 632w\" sizes=\"(max-width: 660px) 100vw, 660px\" />","url":"https://calderaforms.com/wp-content/uploads/edd/2016/11/mark-viewed.png","width":"1200","height":"437","srcset":"https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-300x97.jpg 300w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-768x249.jpg 768w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner-632x205.jpg 632w, https://calderaformscom.lndo.site/wp-content/uploads/2014/09/caldera-forms-banner.jpg 772w","sizes":"(max-width: 150px) 100vw, 150px","alt":"Caldera Forms Banner"},"cfdc_prices":{"ID":20518,"price":"274.89","free":false,"options":[]},"_links":{"self":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download/20518"}],"collection":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download"}],"about":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/types/download"}],"author":[{"embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/users/1"}],"version-history":[{"count":3,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download/20518/revisions"}],"predecessor-version":[{"id":109545,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/download/20518/revisions/109545"}],"wp:featuredmedia":[{"embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/media/16735"}],"wp:attachment":[{"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/media?parent=20518"}],"wp:term":[{"taxonomy":"caldera_admin_help","embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/caldera_admin_help?post=20518"},{"taxonomy":"caldera_doc_overview","embeddable":true,"href":"https://calderaformscom.lndo.site/wp-json/wp/v2/caldera_doc_overview?post=20518"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}}]