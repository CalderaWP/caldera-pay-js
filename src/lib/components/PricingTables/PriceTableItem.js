// @flow
import React from 'react';
import type {Product} from "../../types";
type Props = {
	product: Product,
	onSelectOption: Function,
	features: Array<string>,
	callToAction: string
};
export const PriceTableItem = (props: Props ) => {
	const {product,onSelectOption,features,callToAction} = props;
	const {calderaPay} = product;
	return(
		<div>
			<h3>{product.title.rendered}</h3>
			<ul>
				{features.forEach((feature:string) => {
					return <li>feature</li>
				})}
				<li>Includes {calderaPay.bundle.includes.length} Add-ons</li>
				<li>Includes Caldera Pro Service</li>
			</ul>
			<button
				onClick={onSelectOption}
			>
				{callToAction}
			</button>
		</div>
	)
}

