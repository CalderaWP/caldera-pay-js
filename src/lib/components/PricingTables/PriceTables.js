

// @flow
import React from 'react';
import {PriceTableItem} from "./PriceTableItem";
import type {Product} from "../../types";
type Props = {
	products: Array<Product>,
	callToAction: string,
	onSelectOption: Function
};

/**
 * Display Price table for a component
 *
 * @param props
 * @return {*}
 * @constructor
 */
export const PriceTables = (props: Props ) => {
	const {products,callToAction,onSelectOption} = props;
	return(
		<div>
			{products.map((product: Product) => {
				const {calderaPay} = product;

				const features=  [
					`Unlimited Form Submissions`,
					`Unlimited Forms`
				];
				if(calderaPay.bundle.includes.length ){
					features.push(`Includes ${calderaPay.bundle.includes.length} Add-ons`);
					features.push(`Includes Caldera Pro Service`)
				};

				const onSelectThisOption = () => { onSelectOption(product.id); };
				return (
					<React.Fragment key={product.id}>
						<PriceTableItem
							product={product}
							onSelectOption={onSelectThisOption}
							features={features}
							callToAction={callToAction}
						/>
					</React.Fragment>

				);
			})}
		</div>
	)
};

