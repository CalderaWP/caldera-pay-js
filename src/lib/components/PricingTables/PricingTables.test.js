import {PriceTables} from "./PriceTables";
import {PriceTableItem} from "./PriceTableItem";
import React from 'react';
import renderer from "react-test-renderer";
import bundles from '../../__MOCKDATA__/bundles'

function genericHandler(){}
describe( 'Pricing table components', () => {
	it( 'Item in table snapshot', () => {
		const product = bundles[0];
		const component = renderer.create(<PriceTableItem product={product} onSelectOption={genericHandler} features={['feature1', 'feature 2']} callToAction={'Buy Me!'}/>)
		expect( component.toJSON() ).toMatchSnapshot();
	});
	it( 'Pricing Tables with 3 products', () => {
		const componentWith3Products = renderer.create(
			<PriceTables products={bundles} callToAction={'Buy This ONe'} onSelectOption={genericHandler}/>
		);

		expect( componentWith3Products.toJSON() ).toMatchSnapshot();

	});


	it( 'Pricing Tables with 1 products', () => {

		const product = bundles[0];

		const componentWith1Products = renderer.create(
			<PriceTables products={[product]} callToAction={'Buy the only option'} onSelectOption={genericHandler}/>
		);

		expect( componentWith1Products.toJSON() ).toMatchSnapshot();
	});
})

