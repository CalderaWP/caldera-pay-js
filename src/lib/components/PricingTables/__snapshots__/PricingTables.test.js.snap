// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing table components Item in table snapshot 1`] = `
<div>
  <h3>
    Caldera Forms Pro Agency
  </h3>
  <ul>
    <li>
      Includes 
      31
       Add-ons
    </li>
    <li>
      Includes Caldera Pro Service
    </li>
  </ul>
  <button
    onClick={[Function]}
  >
    Buy Me!
  </button>
</div>
`;

exports[`Pricing table components Pricing Tables with 1 products 1`] = `
<div>
  <div>
    <h3>
      Caldera Forms Pro Agency
    </h3>
    <ul>
      <li>
        Unlimited Form Submissions
      </li>
      <li>
        Unlimited Forms
      </li>
      <li>
        Includes 31 Add-ons
      </li>
      <li>
        Includes Caldera Pro Service
      </li>
    </ul>
    <div>
      549.89
    </div>
    <button
      className={undefined}
      onClick={[Function]}
    >
      Buy the only option
    </button>
  </div>
</div>
`;

exports[`Pricing table components Pricing Tables with 3 products 1`] = `
<div>
  <div>
    <h3>
      Caldera Forms Pro Agency
    </h3>
    <ul>
      <li>
        Unlimited Form Submissions
      </li>
      <li>
        Unlimited Forms
      </li>
      <li>
        Includes 31 Add-ons
      </li>
      <li>
        Includes Caldera Pro Service
      </li>
    </ul>
    <div>
      549.89
    </div>
    <button
      className={undefined}
      onClick={[Function]}
    >
      Buy This ONe
    </button>
  </div>
  <div>
    <h3>
      Caldera Forms Pro Individual
    </h3>
    <ul>
      <li>
        Unlimited Form Submissions
      </li>
      <li>
        Unlimited Forms
      </li>
      <li>
        Includes 10 Add-ons
      </li>
      <li>
        Includes Caldera Pro Service
      </li>
    </ul>
    <div>
      164.89
    </div>
    <button
      className={undefined}
      onClick={[Function]}
    >
      Buy This ONe
    </button>
  </div>
  <div>
    <h3>
      Caldera Forms Pro Advanced
    </h3>
    <ul>
      <li>
        Unlimited Form Submissions
      </li>
      <li>
        Unlimited Forms
      </li>
      <li>
        Includes 15 Add-ons
      </li>
      <li>
        Includes Caldera Pro Service
      </li>
    </ul>
    <div>
      274.89
    </div>
    <button
      className={undefined}
      onClick={[Function]}
    >
      Buy This ONe
    </button>
  </div>
</div>
`;
